tests/scripts/thread-cert/test_srp_auto_host_address.py

```diff
@@ -0,0 +1,234 @@
+        # Form the network.
+
+        client.srp_server_set_enabled(False)
+        client.start()
+        self.simulator.go(15)
+        self.assertEqual(client.get_state(), 'leader')
+
+        server.start()
+        self.simulator.go(5)
+        self.assertEqual(server.get_state(), 'router')
+
+        #-------------------------------------------------------------------
+        # Enable SRP server
+
+        server.srp_server_set_enabled(True)
+        self.simulator.go(5)
+
+        #-------------------------------------------------------------------
+        # Enable auto start mode on SRP client
+
+        self.assertEqual(client.srp_client_get_state(), 'Disabled')
+        client.srp_client_enable_auto_start_mode()
+        self.assertEqual(client.srp_client_get_auto_start_mode(), 'Enabled')
+        self.simulator.go(2)
+
+        self.assertEqual(client.srp_client_get_state(), 'Enabled')
+
+        #-------------------------------------------------------------------
+        # Set host name and enable auto host address on client
+
+        client.srp_client_set_host_name('host')
+        client.srp_client_enable_auto_host_address()
+
+        #-------------------------------------------------------------------
+        # Register a service on client
+
+        client.srp_client_add_service('test_srv', '_test._udo', 12345, 0, 0)
+        self.simulator.go(2)
+        self.check_registered_addresses(client, server)
+
+        #-------------------------------------------------------------------
+        # Add an address and check the SRP client re-registered and updated
+        # server with new address.
+
+        client.add_ipaddr('fd00:1:2:3:4:5:6:7')
+
+        self.simulator.go(5)
+        client_addresses = [addr.strip() for addr in client.get_addrs()]
+        self.assertIn('fd00:1:2:3:4:5:6:7', client_addresses)
+        self.check_registered_addresses(client, server)
+
+        #-------------------------------------------------------------------
+        # Remove the address and check the SRP client re-registered and updated
+        # server.
+
+        client.del_ipaddr('fd00:1:2:3:4:5:6:7')
+
+        self.simulator.go(5)
+        client_addresses = [addr.strip() for addr in client.get_addrs()]
+        self.assertNotIn('fd00:1:2:3:4:5:6:7', client_addresses)
+        self.check_registered_addresses(client, server)
+
+        #-------------------------------------------------------------------
+        # Add an SLAAC on-mesh prefix (which will trigger an address to be
+        # added) and check that the SRP client re-registered and updated
+        # server with the new address.
+
+        client.add_prefix('fd00:abba:cafe:bee::/64', 'paos')
+        client.register_netdata()
+        self.simulator.go(5)
+
+        slaac_addr = [addr.strip() for addr in client.get_addrs() if addr.strip().startswith('fd00:abba:cafe:bee:')]
+        self.assertEqual(len(slaac_addr), 1)
```

Comment:
Add test for auto-adding multiple OMR prefixes when they exist?

Reply:
Good idea. Updated the test in new push to cover situation with multiple addresses.